cpp/src/structs/search/contains.cu

```diff
@@ -35,52 +35,32 @@ bool contains(structs_column_view const& haystack,
               scalar const& needle,
               rmm::cuda_stream_view stream)
 {
-  CUDF_EXPECTS(haystack.type() == needle.type(), "scalar and column types must match");
+  auto const haystack_tv = table_view{{haystack}};
+  // Create a (structs) column_view of one row having children given from the input scalar.
+  auto const needle_tv = static_cast<struct_scalar const*>(&needle)->view();
+  auto const needle_as_col =
+    column_view(data_type{type_id::STRUCT},
+                1,
+                nullptr,
+                nullptr,
+                0,
+                0,
+                std::vector<column_view>{needle_tv.begin(), needle_tv.end()});
```

Comment:
We should probably add overloads of the comparators that take a scalar for the lhs or rhs that do this automaticaly.

Reply:
I'm glad you mentioned this. I was thinking that would be a helpful next step as well, but I hadn't yet spent the time to identify how many times that pattern occurs.

Reply:
This would be an interesting follow up PR.

Reply:
Filed an issue: https://github.com/rapidsai/cudf/issues/10892

Reply:
Also filed a related FEA issue: https://github.com/rapidsai/cudf/issues/10893